src/convertToObject.js

```diff
@@ -6,7 +6,25 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const source = sourceString;
+  const splitedByASemicolon = source
+    .split(';')
+    .map((string) => string.trim())
+    .filter((string) => string.length > 0);
```

Comment:
[CODE KNOWLEDGE] - You don't need to make a copy before each operation. Moreover, methods like `filter` or `map` don't mutate the original array. You can chain these methods directly after the `split` method.